checkov/terraform/plan_parser.py

```diff
@@ -238,14 +239,24 @@ def _get_resource_changes(template: dict[str, Any]) -> dict[str, dict[str, Any]]
     """Returns a resource address to resource changes dict"""
 
     resource_changes_map = {}
-
     resource_changes = template.get("resource_changes")
-    if resource_changes and isinstance(resource_changes, list):
-        resource_changes_map = {
-            change.get("address", ""): change
-            for change in resource_changes
-        }
 
+    if resource_changes and isinstance(resource_changes, list):
+        for each in resource_changes:
+            resource_changes_map[each["address"]] = each
+            changes = []
+
+            # before + after are None when resources are created/destroyed, so make them safe
+            if not each["change"]["before"]:
+                each["change"]["before"] = {}
+            if not each["change"]["after"]:
+                each["change"]["after"] = {}
```

Comment:
`not` also works, but if it is an empty `dict` then there is no need to override it again
```suggestion
            if each["change"]["before"] is None:
                each["change"]["before"] = {}
            if each["change"]["after"] is None:
                each["change"]["after"] = {}
```